src/core/tranche/tranche.sol

```diff
@@ -40,40 +44,58 @@ contract Reserve is DSNote {
     TokenLike public token;
     TokenLike public currency;
 
-    constructor(address token_, address currency_) public {
+    AssessorLike public assessor;
+
+    address public self;
+
+    constructor(address token_, address currency_, address assessor_) public {
         wards[msg.sender] = 1;
+
         token = TokenLike(token_);
         currency = TokenLike(currency_);
+        assessor = AssessorLike(assessor_);
+
+        self = address(this);
     }
 
     function balance() public returns (uint) {
-        return currency.balanceOf(address(this));
+        return currency.balanceOf(self);
     }
 
     function tokenSupply() public returns (uint) {
```

Comment:
idk if this function is necessary, we could just used `token.totalSupply()` directly